wigwam/cli/build_commands.py

```
@@ -170,6 +173,52 @@ def init_build_parsers(subparsers: argparse._SubParsersAction) -> None:
         ' Defaults to "build-installed".',
     )
 
+    parser_build_all = subparsers.add_parser(
+        "build-all",
+        parents=[config_params, no_cache_params],
+        help="Performs the complete compilation process, from initial GitHub checkout "
+        "to installation.",
+        formatter_class=help_formatter,
+    )
+    parser_build_all.add_argument(
+        "--base",
+        "-b",
+        type=str,
+        default="setup-mamba-dev",
+        help='The name of the parent Docker image. Default is "setup-mamba-dev".',
+    )
+    parser_build_all.add_argument(
+        "--tag",
+        "-t",
+        default="build",
+        type=str,
+        help="The sub-prefix of the Docker images to be created. Generated images will "
+        f'have tags fitting "{universal_tag_prefix()}-[TAG]-*". Default: "build"',
+    )
+    parser_build_all.add_argument(
+        "--copy-path",
+        "-p",
+        metavar="FILEPATH",
+        type=str,
+        default=None,
+        help="The path to be copied to the image. If used, no github image will be "
+        "copied. Defaults to None.",
+    )
+    parser_build_all.add_argument(
+        "--archive-url",
+        type=str,
+        metavar="GIT_ARCHIVE",
+        help='The URL of the Git archive to be fetched. Must be a "tar.gz" file.'
+        "Ignored if --copy-path is used.",
```

>Ignored if --copy-path is used.

What happens if the user provides both `--copy-path` AND `--archive-url`? Should we raise an exception to let them know that they've used the interface incorrectly?

One way to handle this is to create a [mutually exclusive parameter group](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_mutually_exclusive_group) and add both `--copy-path` and `--archive-url` to it. That way, `argparse` will ensure that only one of these two options is selected by the user.

If you use `parser.add_mutually_exclusive_group(required=True)`, then the user is required to specify exactly one of the two options.